shapes/u7drag.cc

```diff
@@ -31,20 +31,28 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 #include <cassert>
 
-#define FILE_URI "file:///"
+#define FILE_URI "file://"
 
 /*
  *  Check and skip the URI introducer.
  */
 static bool verify_is_uri(
     const unsigned char *&data,
-	const char *searched_uri
+    const char *searched_uri
 ) {
-	static const size_t urilen = strlen(searched_uri);
-	if (memcmp(data, searched_uri, urilen) == 0) {
+	const size_t urilen = strlen(searched_uri);
+	if (memcmp(data,
+	           searched_uri,
+	           urilen      ) == 0) {
 		data += urilen;
 		return true;
 	}
+	if (memcmp(data,
+	           searched_uri + strlen(FILE_URI),
+	           urilen       - strlen(FILE_URI)) == 0) {
+		data += urilen  - strlen(FILE_URI);
```

Comment:
Can you create a `const size_t` variable for result of the strlen call and use in these 3 locations?

Reply:
I shall immediately.
But don't you think that any reasonable compiler would compute the constant once ?

Reply:
Almost certainly; but it makes for better code anyway.

Reply:
Replacement force pushed.